Gems/ROS2ScriptIntegration/Code/Source/Clients/ROS2ScriptIntegrationSystemComponent.cpp

```diff
@@ -0,0 +1,481 @@
+    {
+        if (ROS2ScriptIntegrationInterface::Get() == this)
+        {
+            ROS2ScriptIntegrationInterface::Unregister(this);
+        }
+        if (PublisherRequesIterface::Get() == this)
+        {
+            PublisherRequesIterface::Unregister(this);
+        }
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::Init()
+    {
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+        ROS2ScriptIntegrationRequestBus::Handler::BusConnect();
+        PublisherRequestBus::Handler::BusConnect();
+        SubscriberRequestBus::Handler::BusConnect();
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::Deactivate()
+    {
+        std::lock_guard<AZStd::mutex> lock1(m_publisherMapMutex);
+        m_publishers.clear();
+        std::lock_guard<AZStd::mutex> lock2(m_subscribersMapMutex);
+        m_subscribers.clear();
+        SubscriberRequestBus::Handler::BusDisconnect();
+        PublisherRequestBus::Handler::BusDisconnect();
+        ROS2ScriptIntegrationRequestBus::Handler::BusDisconnect();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        SubscriberNotificationsBus::Broadcast(&SubscriberNotificationsBus::Events::OnStdMsgBool, "test");
+    }
+
+    // Publishers overrides ...
+    void ROS2ScriptIntegrationSystemComponent::PublishStdMsgsString(const AZStd::string& topicName, const AZStd::string& value)
+    {
+        auto publisher = GetOrCreatePublisher<std_msgs::msg::String>(topicName);
+        if (publisher)
+        {
+            auto message = std_msgs::msg::String();
+            message.data = std::string(value.c_str());
+            publisher->publish(message);
+            return;
+        }
+        return;
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::PublishStdMsgEmpty(const AZStd::string& topicName)
+    {
+        auto publisher = GetOrCreatePublisher<std_msgs::msg::Empty>(topicName);
+        if (publisher)
+        {
+            auto message = std_msgs::msg::Empty();
+            publisher->publish(message);
+            return;
+        }
+        return;
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::PublishStdMsgUInt32(const AZStd::string& topicName, const uint32_t value)
+    {
+        auto publisher = GetOrCreatePublisher<std_msgs::msg::UInt32>(topicName);
+        if (publisher)
+        {
+            auto message = std_msgs::msg::UInt32();
+            message.data = value;
+            publisher->publish(message);
+            return;
+        }
+        return;
+    }
+
+    void ROS2ScriptIntegrationSystemComponent::PublishStdMsgInt32(const AZStd::string& topicName, const int32_t value)
+    {
+        auto publisher = GetOrCreatePublisher<std_msgs::msg::Int32>(topicName);
+        if (publisher)
```

Comment:
Helper method similar to `GetOrCreatePublisher` that could get topic_name and callback would be helpful to simplify duplicate code.